portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +28,49 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function addRandomQuote() {
+    // TODO store quotes in seperate JSON file
+    const greetings =
+      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+
+}
+
+function randomizeImage() {
+  // The images directory contains 11 images, so generate a random index between
+  // 1 and 11.
+  const size = 11;
+  const imageIndex = Math.floor(Math.random() * size) + 1;
+  const imgUrl = 'images/grace-' + imageIndex + '.jpg';
+
+  const imgElement = document.createElement('img');
+  imgElement.src = imgUrl;
+  imgElement.width = '300';
+
+  const imageContainer = document.getElementById('random-image-container');
+  // Remove the previous image.
+  imageContainer.innerHTML = '';
+  imageContainer.appendChild(imgElement);
+
+  // Add Caption
+  var moments = [
+    "text1",
+    "text2",
+    "text3", 
+    "text4",
+    "text5",
+    "text6",
+    "text7",
+    "text8",
+    "text9",
+    "text10",
+    "text11",
+  ]
+  console.log(moments[imageIndex]);
+  const txtElement = document.createElement('p');
+  var txt = document.createTextNode(moments[imageIndex]);
+  txtElement.appendChild(txt);
+  const txtContainer = document.getElementById('random-caption-container');
+  txtContainer.innerHTML = '';
```

Comment:
readability nit: I'd move this line earlier in the function so this function flows sequentially. So it removes the previous children, generates a new image, then adds it.